File: app/src/main/java/com/dimatest/movieapp/database/MovieDao.java
package com.dimatest.movieapp.database;

import androidx.paging.DataSource;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.dimatest.movieapp.database.entity.MovieDO;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

@Dao
public interface MovieDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insertAll(List<MovieDO> movieDOList);

    @Update
    Completable update(MovieDO movieDO);

    @Delete
    Completable delete(MovieDO movieDO);

    @Query("DELETE FROM movies")
    Completable deleteAll();

    @Query("SELECT * FROM movies")
    DataSource.Factory<Integer, MovieDO> getMoviesPaged();

    @Query("SELECT * FROM movies WHERE id =:id")
    Single<MovieDO> getMovieById(long id);
}

File: app/src/main/java/com/dimatest/movieapp/database/MovieDao.kt
package com.dimatest.movieapp.database

import androidx.paging.DataSource
import androidx.room.*
import com.dimatest.movieapp.database.entity.MovieDO

@Dao
interface MovieDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(movieDOList: List<MovieDO>)

    @Update
    suspend fun update(movieDO: MovieDO)

    @Delete
    suspend fun delete(movieDO: MovieDO)

    @Query("DELETE FROM movies")
    suspend fun deleteAll()

    @Query("SELECT * FROM movies")
    fun getMoviesPaged(): DataSource.Factory<Int, MovieDO>

    @Query("SELECT * FROM movies WHERE id =:id")
    suspend fun getMovieById(id: Long): MovieDO
}